### DISCUSSION TITLE
[New challenge] any-str

### DISCUSSION THREAD

**Discussion by F-park** [laike9m/Python-Type-Challenges#90] [Ideas]
## Description

Example from [AnyStr](https://docs.python.org/3/library/typing.html#typing.AnyStr)

## Hint

Check out [AnyStr](https://docs.python.org/zh-cn/3/library/typing.html#typing.AnyStr)

## Code

```py
"""
TODO:

Define an `AnyStr` type that accept str or bytes arguments but cannot allow the two to mix.
"""

from typing import TypeVar

AnyStr = TypeVar("AnyStr", str, bytes)


## End of your code ##
def concat(a: AnyStr, b: AnyStr) -> AnyStr:
    return a + b


concat("foo", "bar")
concat(b"foo", b"bar")
concat("foo", b"bar")  # expect-type-error
```

**Comment by laike9m** [comment]
Thanks for contributing this idea.

I can think of two ways to improve the challenge:
a. Show how to use AnyStr, instead of letting users define it.
b. Still use the idea of creating a generalized custom type, but tweak the requirement a little (or with completely new requirements unrelated to strings and bytes) to not duplicate what's already in the doc

Both work for me
